services/brig/src/Brig/API/User.hs

```diff
@@ -183,7 +183,16 @@ createUser new = do
   (email, phone) <- validateEmailAndPhone new
 
   -- get invitation and existing account
-  (newTeam, teamInvitation, tid) <- handleTeam (newUserTeam new) (userEmailKey <$> email)
+  (newTeam, teamInvitation, tid) <-
+    case (newUserTeam new, userEmailKey <$> email) of
```

Comment:
dito: why inline some, and abtract others?

Reply:
To me this function is interesting business logic that I'd rather see in the body. All other functions in the where block are not that interesting business logic-wise and are okay to abstract behind a name.